index.bs

```diff
@@ -7055,11 +7049,26 @@ an |ownership type|, a |serialization internal map|, a |realm| and a |session|:
         production in the [=local end definition=], with the <code>handle</code>
         property set to |handle id| if it's not null, or omitted otherwise.
 
+    <dt>|value| has a \[[SyncIteratorRecord]] [=internal slot=]
```

Comment:
This is for a sync iterators only but we have to support async ones as well. Not sure if both could be merged here or if that would require separate types? @jgraham might be better to comment on that.

Reply:
As far as I understood from here: https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-createasyncfromsynciterator, async iterators should also have `SyncIteratorRecord` internal slot.

Reply:
I think this is wrong either way; the specification can create async iterators from sync iterators by creating objects with `[[SyncIteratorRecord]]` internal slots, but a [general iterator](https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-iterator-interface) only has to have a `next` method. But maybe we want to check for `%IteratorPrototype%` or `%AsyncIteratorPrototype%` on the prototype chain?

Reply:
Maksym in [this issue](https://github.com/w3c/webdriver-bidi/issues/537) also suggested checking if `next` method is callable for iterators. @sadym-chromium, could you maybe elaborate on your suggestion?

Reply:
Alright, I've tried to update the check for iterators in the terms of inheritance from `%IteratorPrototype%` or `%AsyncIteratorPrototype%`. Let me know what you think.

Reply:
I'm quite sceptical about calling an `iterator` any object with callable property `next`.
I'd question if we have real-world scenarios, where iterators serialization is required, and if not - just consider it's an object.